src/layout/TreasurySummaryTeaser.tsx

```diff
@@ -7,17 +7,19 @@ import SeactionTeaserContainer from 'presentational/SectionTeaserContainer';
 import {useBlockTime} from 'src/api/hooks/useBlockTime';
 import ProgressChartWidget from 'presentational/ProgressWidget';
 import StatInfoBlock from 'presentational/StatInfoBlock';
-import withTreasury, {InjectedPropTypes} from 'src/hoc/withTreasury';
 import {formatNumber, BN_ONE, bnToBn, formatBalance, BN_ZERO} from '@polkadot/util';
+import {useTreasuryInfo} from 'src/api/hooks/useTreasuryInfo';
+import {useBestNumber} from 'src/api/hooks/useBestNumber';
 
 type PropTypes = {
   onPressMore: () => void;
 };
 
-function TreasurySummaryTeaser(props: PropTypes & InjectedPropTypes) {
-  const {treasuryInfo} = props;
-  const total = treasuryInfo.spendPeriod || BN_ONE;
-  const value = treasuryInfo.bestNumber?.mod(treasuryInfo.spendPeriod?.toBn() ?? BN_ONE);
+export function TreasurySummaryTeaser(props: PropTypes) {
+  const bestNumber = useBestNumber();
+  const {data: treasuryInfo} = useTreasuryInfo();
```

Comment:
Is there any loading state? the entire component depends on `treasuryInfo`, it would probably make sense to return early when its `undefined`

Reply:
done ✅